File: eval/fixtures/php-taskflow/src/Factory.php
<?php

declare(strict_types=1);

namespace App;

use App\Http\Handler;
use App\Service\Service;
use App\Store\MemStore;
use App\Store\Store;

/** Composition root: wires the concrete store up through the HTTP handler. */
final class Factory
{
    public static function build(): Handler
    {
        $store = self::store();
        return new Handler(new Service($store));
    }

    public static function store(): Store
    {
        return new MemStore();
    }
}

File: eval/fixtures/python-taskflow/taskflow/factory.py
"""Composition root: wires the concrete store up through the HTTP handler."""

from taskflow.http.handler import Handler
from taskflow.service.service import Service
from taskflow.store.base import Store
from taskflow.store.memstore import MemStore


def build() -> Handler:
    return Handler(Service(store()))


def store() -> Store:
    return MemStore()
